### ISSUE TITLE
OCI runtime exec failed: exec failed: unable to start container process: exec: "node": executable file not found in $PATH: unknown

### ISSUE THREAD

**Issue by DSOE1024** [abarichello/godot-ci#140]
OCI runtime exec failed: exec failed: unable to start container process: exec: "node": executable file not found in $PATH: unknown

One more serious problem was that I reinstalled nodejs and dotnet but still couldn't build successfully

Class 'WindowWrapper' is not exposed, skipping.
Loaded system CA certificates
EditorSettings: Save OK!
ERROR: Parent node is busy setting up children, `add_child()` failed. Consider using `add_child.call_deferred(child)` instead.
   at: add_child (scene/main/node.cpp:1415)
ERROR: Condition "!is_inside_tree()" is true.
   at: popup_centered (scene/main/window.cpp:1658)

**Comment by TheLongFellowPLAYER** [comment]
Heads up, also got this issue, granted I'm using act runners. However, looks like the base Godot-CI docker image does not possess nodejs. Switching the godot-ci base to nodejs 20 bullseye appears to have resolved that issue, however builds are failing with "This project doesn't have an 'export_presets.cfg" file at it's root." Still troubleshooting.
Dockerfile:
https://pastebin.com/ZCfuQQpg

**Comment by TheLongFellowPLAYER** [comment]
Yeah, that's the actions file, I'm specifically referring to building the docker image based off of Node20 Bullseye. 
`FROM node:20-bullseye
LABEL author="https://github.com/aBARICHELLO/godot-ci/graphs/contributors"
 
USER root
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update && apt-get install -y --no-install-recommends \
    ca-certificates \
    git \
    git-lfs \
    unzip \
    wget \
    zip \
    adb \
    openjdk-17-jdk-headless \
    rsync \
    && rm -rf /var/lib/apt/lists/*
 
ARG GODOT_VERSION="4.2.1"
ARG RELEASE_NAME="stable"
ARG SUBDIR=""
ARG GODOT_TEST_ARGS=""
ARG GODOT_PLATFORM="linux.x86_64"
 
RUN wget https://downloads.tuxfamily.org/godotengine/${GODOT_VERSION}${SUBDIR}/Godot_v${GODOT_VERSION}-${RELEASE_NAME}_${GODOT_PLATFORM}.zip \
    && wget https://downloads.tuxfamily.org/godotengine/${GODOT_VERSION}${SUBDIR}/Godot_v${GODOT_VERSION}-${RELEASE_NAME}_export_templates.tpz \
    && mkdir ~/.cache \
    && mkdir -p ~/.config/godot \
    && mkdir -p ~/.local/share/godot/export_templates/${GODOT_VERSION}.${RELEASE_NAME} \
    && unzip Godot_v${GODOT_VERSION}-${RELEASE_NAME}_${GODOT_PLATFORM}.zip \
    && mv Godot_v${GODOT_VERSION}-${RELEASE_NAME}_${GODOT_PLATFORM} /usr/local/bin/godot \
    && unzip Godot_v${GODOT_VERSION}-${RELEASE_NAME}_export_templates.tpz \
    && mv templates/* ~/.local/share/godot/export_templates/${GODOT_VERSION}.${RELEASE_NAME} \
    && rm -f Godot_v${GODOT_VERSION}-${RELEASE_NAME}_export_templates.tpz Godot_v${GODOT_VERSION}-${RELEASE_NAME}_${GODOT_PLATFORM}.zip
 
ADD getbutler.sh /opt/butler/getbutler.sh
RUN bash /opt/butler/getbutler.sh
RUN /opt/butler/bin/butler -V
 
ENV PATH="/opt/butler/bin:${PATH}"
 
# Download and setup android-sdk
ENV ANDROID_HOME="/usr/lib/android-sdk"
RUN wget https://dl.google.com/android/repository/commandlinetools-linux-7583922_latest.zip \
    && unzip commandlinetools-linux-*_latest.zip -d cmdline-tools \
    && mv cmdline-tools $ANDROID_HOME/ \
    && rm -f commandlinetools-linux-*_latest.zip
 
ENV PATH="${ANDROID_HOME}/cmdline-tools/cmdline-tools/bin:${PATH}"
 
RUN yes | sdkmanager --licenses \
    && sdkmanager "platform-tools" "build-tools;33.0.2" "platforms;android-33" "cmdline-tools;latest" "cmake;3.22.1" "ndk;25.2.9519653"
 
# Adding android keystore and settings
RUN keytool -keyalg RSA -genkeypair -alias androiddebugkey -keypass android -keystore debug.keystore -storepass android -dname "CN=Android Debug,O=Android,C=US" -validity 9999 \
    && mv debug.keystore /root/debug.keystore
 
RUN godot -v -e --quit --headless ${GODOT_TEST_ARGS}
RUN echo 'export/android/android_sdk_path = "/usr/lib/android-sdk"' >> ~/.config/godot/editor_settings-4.tres
RUN echo 'export/android/debug_keystore = "/root/debug.keystore"' >> ~/.config/godot/editor_settings-4.tres
RUN echo 'export/android/debug_keystore_user = "androiddebugkey"' >> ~/.config/godot/editor_settings-4.tres
RUN echo 'export/android/debug_keystore_pass = "android"' >> ~/.config/godot/editor_settings-4.tres
RUN echo 'export/android/force_system_user = false' >> ~/.config/godot/editor_settings-4.tres
RUN echo 'export/android/timestamping_authority_url = ""' >> ~/.config/godot/editor_settings-4.tres
RUN echo 'export/android/shutdown_adb_on_exit = true' >> ~/.config/godot/editor_settings-4.tres`

**Comment by TheLongFellowPLAYER** [comment]
@DSOE1024 Unfortunately still not functional for me, however would attempt building this docker file, has mono-complete as a part of build process. Might work, no idea though.
Docker File:
https://pastebin.com/fm0A3GAQ

**Comment by weightedangelcube** [comment]
Related to nektos/act#973. Action should work if you upload to GitHub.
